model-optimizer/extensions/middle/UpsampleToResample.py

```diff
@@ -92,10 +92,10 @@ def replace_pattern(self, graph: Graph, match: Dict[str, Node]):
 
         if input_shape_rank == 4:
             begin_value = int64_array([get_height_dim(layout, input_shape_rank)])
-            factor_value = np.array([height_scale, width_scale])
+            factor_value = int64_array([height_scale, width_scale])
         else:
             begin_value = int64_array([get_depth_dim(layout, input_shape_rank)])
-            factor_value = np.array([depth_scale, height_scale, width_scale])
+            factor_value = int64_array([depth_scale, height_scale, width_scale])
```

Comment:
This is wrong change. After the change the `Mul` operation gets a int64 constant in port 1 and floating point constant in port 0 (refer to line 155 where `convert_to_float` operation is inserted). 

Moreover, the `height_scale` and `width_scale` are of floating point type. Refer to the `model-optimizer/extensions/front/onnx/upsample_ext.py`

Reply:
Update unit tests for the transformation and make sure that proper data types are used. Also, add a test where `height_scale` and `width_scale` are not equal to some integer, for example, '3.5'

Reply:
Fixed.